Loader/Browser Thumbnail improvements: Double-Click to Open & video representation thumbnail playback

FuzzkingCool opened the pull request:
This pull request enhances the thumbnail preview functionality in the Loader tool, introducing support for video thumbnails (particularly h264_* and other common video formats) and enabling video playback directly within the thumbnail widget. It also allows users to open thumbnails with the system's default viewer via double-click.     

**Video thumbnail and playback support:**

* Added logic to detect video representations (h264_*, mp4, mov, etc.) for selected versions and display their first frame as a thumbnail. If a video is detected, the widget shows a play overlay and supports playback using QtMultimedia if available. (`client/ayon_core/tools/loader/ui/window.py`, `client/ayon_core/tools/utils/thumbnail_paint_widget.py`) [[1]](diffhunk://#diff-8267e7c7796b41a9fba43ac49f913575391b96dbf809fa03ea575a9fe5734cdaR634-R729) [[2]](diffhunk://#diff-b1c81216558553ee139ee3471ec78d749d466d33a9cf27512cd7c326a39b74eaR50-R179) [[3]](diffhunk://#diff-b1c81216558553ee139ee3471ec78d749d466d33a9cf27512cd7c326a39b74eaL113-R295)
* Implemented extraction of the first frame from video files using ffmpeg for preview purposes. (`client/ayon_core/tools/utils/thumbnail_paint_widget.py`)

**User interaction improvements:**

* Added double-click support on thumbnails to open the image or video with the system's default viewer, handling cross-platform compatibility. (`client/ayon_core/tools/loader/ui/window.py`, `client/ayon_core/tools/utils/thumbnail_paint_widget.py`) [[1]](diffhunk://#diff-8267e7c7796b41a9fba43ac49f913575391b96dbf809fa03ea575a9fe5734cdaR222-R225) [[2]](diffhunk://#diff-8267e7c7796b41a9fba43ac49f913575391b96dbf809fa03ea575a9fe5734cdaR755-R770) [[3]](diffhunk://#diff-b1c81216558553ee139ee3471ec78d749d466d33a9cf27512cd7c326a39b74eaR31-R33)

**UI and code refactoring:**

* Improved painting logic and code formatting for thumbnail rendering, including overlay drawing and aspect ratio handling. (`client/ayon_core/tools/utils/thumbnail_paint_widget.py`) [[1]](diffhunk://#diff-b1c81216558553ee139ee3471ec78d749d466d33a9cf27512cd7c326a39b74eaR305-R380) [[2]](diffhunk://#diff-b1c81216558553ee139ee3471ec78d749d466d33a9cf27512cd7c326a39b74eaL171-R404) [[3]](diffhunk://#diff-b1c81216558553ee139ee3471ec78d749d466d33a9cf27512cd7c326a39b74eaL192-R432) [[4]](diffhunk://#diff-b1c81216558553ee139ee3471ec78d749d466d33a9cf27512cd7c326a39b74eaL225-R454) [[5]](diffhunk://#diff-b1c81216558553ee139ee3471ec78d749d466d33a9cf27512cd7c326a39b74eaL248-R470) [[6]](diffhunk://#diff-b1c81216558553ee139ee3471ec78d749d466d33a9cf27512cd7c326a39b74eaL331-R551)

iLLiCiTiT committed:
Merge branch 'develop' into bugfix/blind-flickering-fix

iLLiCiTiT committed:
move addons manager to controller

iLLiCiTiT committed:
Merge pull request #1358 from ynput/bugfix/blind-flickering-fix

Publisher: Try to fix flickering issue in publisher report

iLLiCiTiT committed:
added workfile items to launcher controller

iLLiCiTiT committed:
added workfiles page to launcher window

iLLiCiTiT committed:
use standard proxy model

iLLiCiTiT committed:
formatting fixes

iLLiCiTiT committed:
added 'get_project_entity' to controller

iLLiCiTiT committed:
add icons mapping to products

iLLiCiTiT committed:
use same deafult as backend has

iLLiCiTiT committed:
move 'ProductTypeIconMapping' to common project model

iLLiCiTiT committed:
use product type icon in scene inventory

iLLiCiTiT committed:
remove unused import

iLLiCiTiT committed:
use icon color from project anatomy for tasks

iLLiCiTiT committed:
elite status name

iLLiCiTiT committed:
Merge pull request #1438 from ynput/enhancement/1437-task-icon-color-in-tools

Tasks widget: Use icon color from project anatomy

iLLiCiTiT committed:
Merge branch 'develop' into enhancement/1287-ay-7809_icons-and-colorcoding-for-product-types-in-browser

iLLiCiTiT committed:
Merge pull request #1435 from ynput/enhancement/1287-ay-7809_icons-and-colorcoding-for-product-types-in-browser

Loader: Product type icons

iLLiCiTiT committed:
Merge branch 'develop' into enhancement/elide-status-name-text

iLLiCiTiT committed:
Merge pull request #1439 from ynput/enhancement/elide-status-name-text

Loader: Elide status name text

iLLiCiTiT committed:
fill task type color in publisher's tasks enum

iLLiCiTiT committed:
Merge pull request #1440 from ynput/enhancement/task-type-color-in-publisher

Publisher: Fill task type color in tasks enum

MustafaJafar committed:
Merge branch 'develop' into feature/ci-mkdocs

MustafaJafar committed:
MK Docs: Update light/dark modes icons

MustafaJafar committed:
move mkdocs_requirements to root location.

MustafaJafar committed:
CI: Re-use `deploy_mkdocs` from ynput/ops-repo-automation

MustafaJafar committed:
CI: Update Deploy MkDocs to use action from develop branch from ynput/ops-repo-automation

MustafaJafar committed:
Merge pull request #1441 from ynput/feature/ci-mkdocs-with-enhancements

Feature/CI MK Docs Deployment

iLLiCiTiT committed:
allow explicit workfile path to be defined

iLLiCiTiT committed:
added workfile selection and actions

iLLiCiTiT committed:
Merge branch 'develop' into feature/1229-ay-1284_show-application-used-on-task

iLLiCiTiT committed:
allow to implement 'get_app_information' by host

iLLiCiTiT committed:
allow to pass 'data' to save workfile info

iLLiCiTiT committed:
add app information to workfile data

iLLiCiTiT committed:
remove unused import

iLLiCiTiT committed:
use f-string

iLLiCiTiT committed:
keep ayon application full name as is

iLLiCiTiT committed:
use ayon prefix for ayon app name and tools

ynbot committed:
[Automated] Add generated package files from main

ynbot committed:
[Automated] Update version in package.py for develop

github-actions[bot] committed:
chore(): update bug report / version

iLLiCiTiT committed:
Merge branch 'develop' into feature/1229-ay-1284_show-application-used-on-task

iLLiCiTiT committed:
Merge pull request #1434 from ynput/feature/1229-ay-1284_show-application-used-on-task

Launcher: Show task workfiles

iLLiCiTiT committed:
small changes or logic order

iLLiCiTiT committed:
do not consider locked containers in 'get_outdated_containers' as outdated

iLLiCiTiT committed:
look for locked version in container

iLLiCiTiT committed:
store the information to the model item

iLLiCiTiT committed:
allow more options for icons

iLLiCiTiT committed:
draw a lock next to version if is locked

iLLiCiTiT committed:
make sure 'data_changed' is triggered

iLLiCiTiT committed:
use Logger as log attribute for loader plugin

iLLiCiTiT committed:
remove commented line

Co-authored-by: Roy Nieterau <roy_nieterau@hotmail.com>

iLLiCiTiT committed:
allow to ignore locked versions

iLLiCiTiT committed:
added typehints

iLLiCiTiT committed:
add a todo to 'filter_containers'

iLLiCiTiT committed:
revert back output handling

kalisp committed:
Fix wrong key

iLLiCiTiT committed:
Merge pull request #1447 from ynput/enhancement/l1446-version-pinning

Load: Simple version lock on container

iLLiCiTiT committed:
always fill ayon app name and tools

iLLiCiTiT committed:
Merge branch 'develop' into enhancement/use-ayon-logger-in-load-plugins

iLLiCiTiT committed:
Merge pull request #1449 from ynput/enhancement/use-ayon-logger-in-load-plugins

Load: Use Logger in loader plugin

iLLiCiTiT committed:
Merge branch 'develop' into enhancement/1443-allow-host-addons-to-store-dcc-version-in-workfile-metadata

iLLiCiTiT committed:
Merge pull request #1445 from ynput/enhancement/1443-allow-host-addons-to-store-dcc-version-in-workfile-metadata

Workfile info: Add application information to workfile entity

iLLiCiTiT committed:
use correct key to get action label

iLLiCiTiT committed:
Merge pull request #1452 from ynput/bugfix/webactions-label-handling

Launcher: Fix form handling

iLLiCiTiT committed:
remove milestone versions

iLLiCiTiT committed:
use f-string

iLLiCiTiT committed:
add type hints

iLLiCiTiT committed:
Merge pull request #1455 from ynput/enhancement/addons-manager-api-update

Chore: Addons manager update

iLLiCiTiT committed:
use ayon api implementation of mime type

iLLiCiTiT committed:
copy implementation from ayon api

iLLiCiTiT committed:
added comments to safe import

iLLiCiTiT committed:
Merge pull request #1457 from ynput/enhancement/mime-type-from-ayon-api

Chore: Use ayon api implementation of mime type

philippe-ynput committed:
chore(deps): update Material Symbols to version 4.0.0

Updated Material Symbols font resources to version 4.0.0, including renaming font files, updating codepoints and JSON mappings, and modifying the initialization script to reference the new font name.

BREAKING CHANGE: Font name changed from "MaterialSymbolsOutlined-Regular" to "MaterialSymbolsOutlined", which may affect code relying on the previous naming convention.

Signed-off-by: Philippe Leprince <philippe@ynput.io>

iLLiCiTiT committed:
Merge pull request #1461 from ynput/Update-material-symbols-to-version-4.0.0

chore(deps): update Material Symbols to version 4.0.0

iLLiCiTiT committed:
override '_drawtext'  in ffmpeg burnins

ynbot committed:
[Automated] Add generated package files from main

ynbot committed:
[Automated] Update version in package.py for develop

github-actions[bot] committed:
chore(): update bug report / version

iLLiCiTiT committed:
fix import of otio burnins

iLLiCiTiT committed:
Merge pull request #1463 from ynput/bugfix/otio-adapters-import

Burnins: Fix import of adapters

ynbot committed:
[Automated] Add generated package files from main

ynbot committed:
[Automated] Update version in package.py for develop

github-actions[bot] committed:
chore(): update bug report / version

iLLiCiTiT committed:
revert otio to use older version

iLLiCiTiT committed:
use better method name

iLLiCiTiT committed:
remove irrelevant log

iLLiCiTiT committed:
fallback to oiio if ffmpeg conversion fails

iLLiCiTiT committed:
use review channels to create the thumbnail

iLLiCiTiT committed:
allow 'rgba' layer name to be used if is available

iLLiCiTiT committed:
avoid duplicated code

iLLiCiTiT committed:
grammar fix

Co-authored-by: Roy Nieterau <roy_nieterau@hotmail.com>

iLLiCiTiT committed:
add warning logs if thumbnail creation fails

iLLiCiTiT committed:
Merge pull request #1471 from ynput/bugfix/fallback-to-oiio-if-ffmpeg-fails

Thumbnail: Fallback to OIIO if ffmpeg fails

iLLiCiTiT committed:
Merge branch 'develop' into bugfix/burnins-pillow-update-fix

iLLiCiTiT committed:
Merge pull request #1464 from ynput/bugfix/burnins-pillow-update-fix

Burnins: Support newer Pillow

iLLiCiTiT committed:
Merge branch 'develop' into bugfix/revert-to-older-OTIO

iLLiCiTiT committed:
Merge pull request #1466 from ynput/bugfix/revert-to-older-OTIO

Chore: Revert OTIO to use older version

ynbot committed:
[Automated] Add generated package files from main

ynbot committed:
[Automated] Update version in package.py for develop

github-actions[bot] committed:
chore(): update bug report / version

sosoyan committed:
Fix typo

iLLiCiTiT committed:
Merge pull request #1475 from sosoyan/develop

Fix typo in "getbbox" attribute

iLLiCiTiT committed:
just set minimum size

iLLiCiTiT committed:
use py3 super

kalisp committed:
Extracted gap filling from last version

kalisp committed:
Push through reuseLastVersion value to deadline metadata

kalisp committed:
Return attrib for frameStart query

kalisp committed:
Removed confusing message

kalisp committed:
Moved message where it makes sense

kalisp committed:
Ruff

kalisp committed:
Updated typin

Co-authored-by: Jakub Trllo <43494761+iLLiCiTiT@users.noreply.github.com>

kalisp committed:
Removed unneeded import

Co-authored-by: Jakub Trllo <43494761+iLLiCiTiT@users.noreply.github.com>

kalisp committed:
Removed unnecessary parentheses

Co-authored-by: Jakub Trllo <43494761+iLLiCiTiT@users.noreply.github.com>

kalisp committed:
Formatting change

Co-authored-by: Jakub Trllo <43494761+iLLiCiTiT@users.noreply.github.com>

kalisp committed:
Removed import

Co-authored-by: Jakub Trllo <43494761+iLLiCiTiT@users.noreply.github.com>

kalisp committed:
Added return type

Co-authored-by: Jakub Trllo <43494761+iLLiCiTiT@users.noreply.github.com>

kalisp committed:
Fixed return

Co-authored-by: Jakub Trllo <43494761+iLLiCiTiT@users.noreply.github.com>

iLLiCiTiT committed:
Merge pull request #1479 from ynput/bugfix/avoid-using-qapp-desktop

UI: Avoid using QApplication desktop

kalisp committed:
Moved fill_sequence_gaps_with_previous

kalisp committed:
Renamed fill_sequence_gaps_with_previous

kalisp committed:
Formatting change

Co-authored-by: Jakub Trllo <43494761+iLLiCiTiT@users.noreply.github.com>

iLLiCiTiT committed:
Merge branch 'develop' into feature/YN-0079_fix_frames_on_deadline

kalisp committed:
Reorder imports

Co-authored-by: Jakub Trllo <43494761+iLLiCiTiT@users.noreply.github.com>

kalisp committed:
Ruff

kalisp committed:
Ruff

jakubjezek001 committed:
Updates review extraction process.

Refactors review extraction to remove
Ftrack-specific elements.

- Removes redundant data related to frame
  starts in Ftrack.
- Simplifies review representation.
- Streamlines the audio input processing.

iLLiCiTiT committed:
use settings summary to resolve which addon versions are used

antirotor committed:
🪲 get the correct template for resources

antirotor committed:
🐕 fix linting issues

kalisp committed:
Merge develop

antirotor committed:
♻️use collected data instead

antirotor committed:
♻️ and remove unnecessary import

antirotor committed:
🪲 fix task and product type

antirotor committed:
🦗 fix data access

antirotor committed:
🦗 fix key name

antirotor committed:
Apply suggestion from @iLLiCiTiT

Co-authored-by: Jakub Trllo <43494761+iLLiCiTiT@users.noreply.github.com>

jakubjezek001 committed:
Fixes source metadata key in OTIO review extraction

The change corrects the metadata key prefixes used
when extracting source width and height
information for OTIO reviewable representations.

It removes the trailing period from the prefixes
"ayon.source." and "openpype.source." to ensure
accurate retrieval of resolution data. This resolves
an issue where incorrect or missing resolution
information could lead to squished reviewables.

antirotor committed:
Merge pull request #1487 from ynput/bugfix/respect-templates-for-resources

Fix: Get the correct template for resources

kalisp committed:
Merge branch 'develop' into feature/YN-0079_fix_frames_on_deadline

kalisp committed:
Ruff

kalisp committed:
Merge branch 'feature/YN-0079_fix_frames_on_deadline' of https://github.com/ynput/ayon-core into feature/YN-0079_fix_frames_on_deadline

kalisp committed:
Merge pull request #1480 from ynput/feature/YN-0079_fix_frames_on_deadline

Feature: Fix frames on Deadline

iLLiCiTiT committed:
Merge branch 'develop' into enhancement/removing-ftrack-related-code-extract-review

iLLiCiTiT committed:
Merge pull request #1484 from ynput/enhancement/removing-ftrack-related-code-extract-review

Refactors review extraction process

iLLiCiTiT committed:
better typehints

iLLiCiTiT committed:
Implemented helper dict to handle str -> dict conversion

iLLiCiTiT committed:
rename 'DefaultValueDict' to 'DefaultKeysDict'

iLLiCiTiT committed:
added helper function to get ayon user data

iLLiCiTiT committed:
fill user data as dictionary

MustafaJafar committed:
Revert changes in `pyproject.toml` from #1215

MustafaJafar committed:
Merge branch 'develop' into bugfix/1245-default-render-product-name-profile-is-not-unique-for-each-aov

MustafaJafar committed:
Merge pull request #1246 from ynput/bugfix/1245-default-render-product-name-profile-is-not-unique-for-each-aov

add optional `aov` key to render template profile name

iLLiCiTiT committed:
Merge branch 'develop' into chore/revert-mkdocs-changes-in-pyproject-toml

iLLiCiTiT committed:
Merge pull request #1497 from ynput/chore/revert-mkdocs-changes-in-pyproject-toml

Revert mkdocs changes in `pyproject.toml`

iLLiCiTiT committed:
Merge branch 'develop' into bugfix/addon-versions-resolution

iLLiCiTiT committed:
Merge pull request #1486 from ynput/bugfix/addon-versions-resolution

Bundle: Project bundle addons version resolution

ynbot committed:
[Automated] Add generated package files from main

ynbot committed:
[Automated] Update version in package.py for develop

github-actions[bot] committed:
chore(): update bug report / version

jakubjezek001 committed:
Fixes: Corrects file sequence frame offset

Corrects the calculation of the frame offset
for file sequences in editorial workflows.

- Ensures accurate frame mapping.
- Resolves issues with incorrect frame ranges.

jakubjezek001 committed:
Fixes: Corrects reviewable output resolution.

Updates the expected resolution in the ffmpeg commands
used in reviewable extraction tests to match the
intended output.

This resolves a squashed reviewables issue where the output
resolution was incorrect.

jakubjezek001 committed:
Merge branch 'develop' into bugfix/YN-0123-hiero-squished-reviewables

iLLiCiTiT committed:
pass studio bundle name to farm

iLLiCiTiT committed:
Merge pull request #1499 from ynput/bugfix/studio-bundle-name-on-farm

Farm: Studio bundle name to farm

iLLiCiTiT committed:
use user entity when getting template data during publishing

iLLiCiTiT committed:
allow to pass user entity to get_template_data

iLLiCiTiT committed:
custom handling of 'user' data used for template

iLLiCiTiT committed:
use same logic in integrate hero as in integrate

iLLiCiTiT committed:
add docstring

kalisp committed:
Typing

At least some, dont know how to import NewFolderDict

kalisp committed:
Check that source folder type could be pushed to destination

kalisp committed:
Copy source folder type to destination

ynbot committed:
[Automated] Add generated package files from main

ynbot committed:
[Automated] Update version in package.py for develop

github-actions[bot] committed:
chore(): update bug report / version

jakubjezek001 committed:
Merge branch 'develop' into bugfix/YN-0123-hiero-squished-reviewables

jakubjezek001 committed:
Merge pull request #1489 from ynput/bugfix/YN-0123-hiero-squished-reviewables

Fixes source metadata key for OTIO review

jakubjezek001 committed:
Merge branch 'develop' into bugfix/YN-0080-frame-range-is-showing-wrong-in-publisher

kalisp committed:
Create destination task if no task selected

iLLiCiTiT committed:
Merge branch 'develop' into bugfix/YN-0133_copy_folder_type

kalisp committed:
Extracted logic to methods

kalisp committed:
Typing

iLLiCiTiT committed:
remove custom copy and deepcopy implementation

iLLiCiTiT committed:
Merge branch 'develop' into enhancement/1496-yn-0069-more-user-data-in-templates

kalisp committed:
Fix use operations instead of ayon_api

Must be in same session as create folder if 'Create New Folder'

kalisp committed:
Fix typing

Co-authored-by: Jakub Trllo <43494761+iLLiCiTiT@users.noreply.github.com>

kalisp committed:
Change return of _check_src_folder_type

Co-authored-by: Jakub Trllo <43494761+iLLiCiTiT@users.noreply.github.com>

kalisp committed:
Simplify pop

Co-authored-by: Roy Nieterau <roy_nieterau@hotmail.com>

kalisp committed:
Update usage of renamed _get_dst_folder_type

kalisp committed:
Fix dict typing

iLLiCiTiT committed:
Merge pull request #1501 from ynput/enhancement/1496-yn-0069-more-user-data-in-templates

Template data: More user data in templates

kalisp committed:
Merge branch 'develop' into bugfix/YN-0133_copy_folder_type

kalisp committed:
Merge pull request #1502 from ynput/bugfix/YN-0133_copy_folder_type

Library: copy folder type

kalisp committed:
Use lower only for comparison

Co-authored-by: Jakub Trllo <43494761+iLLiCiTiT@users.noreply.github.com>

kalisp committed:
Remove unnecessary variable

Co-authored-by: Jakub Trllo <43494761+iLLiCiTiT@users.noreply.github.com>

kalisp committed:
Formatting change

Co-authored-by: Jakub Trllo <43494761+iLLiCiTiT@users.noreply.github.com>

kalisp committed:
Merge develop

kalisp committed:
Removed hard fail, unnecessary

kalisp committed:
Added validation that task type is in destination project

kalisp committed:
Merge origin

jakubjezek001 committed:
Merge branch 'develop' into bugfix/YN-0080-frame-range-is-showing-wrong-in-publisher

kalisp committed:
Fix use of lower task name

kalisp committed:
Fix if source version doesn't have task

Co-authored-by: Jakub Trllo <43494761+iLLiCiTiT@users.noreply.github.com>

kalisp committed:
Fix overwriting real task name with name of task type

kalisp committed:
Return task info from created object

kalisp committed:
Task is optional

kalisp committed:
Fix pushed products not attaching to version

kalisp committed:
Merge branch 'enhancement/YN-0132_create_source_task_if_no_selected' of https://github.com/ynput/ayon-core into enhancement/YN-0132_create_source_task_if_no_selected

kalisp committed:
Transfer status and tags

kalisp committed:
Fix access to dict keys

kalisp committed:
Fix access to dict keys

kalisp committed:
Do not send dummy status if not found

kalisp committed:
Merge pull request #1503 from ynput/enhancement/YN-0132_create_source_task_if_no_selected

Library: copy and create source task if no task selected

jakubjezek001 committed:
Handles missing media references in OTIO clips

Adds a check for missing media references in OTIO clips
during publishing.

jakubjezek001 committed:
Merge branch 'develop' into bugfix/YN-0073_publishing-clip-without-media-source-link

BigRoy committed:
Use correct publish template in `get_instance_expected_output_path`

jakubjezek001 committed:
Merge pull request #1505 from ynput/bugfix/YN-0073_publishing-clip-without-media-source-link

Handles missing OTIO media references

iLLiCiTiT committed:
fix mytasks filtering

iLLiCiTiT committed:
lock pyobjc-core to 11.1

iLLiCiTiT committed:
Merge branch 'develop' into bugfix/1504-yn-0141-my-tasks-not-displaying-in-workfiles-app-inside-blender-host

iLLiCiTiT committed:
Remove dev print

MustafaJafar committed:
Merge branch 'develop' into 1488-yn-0117-usd-texture-publish-to-different-drive

kalisp committed:
Implemented copy of source folder thumbnail

iLLiCiTiT committed:
don't require 'AYON_STUDIO_BUNDLE_NAME' to be set

iLLiCiTiT committed:
Merge pull request #1508 from ynput/bugfix/1504-yn-0141-my-tasks-not-displaying-in-workfiles-app-inside-blender-host

Workfiles tool: Fix my tasks filtering

kalisp committed:
Renamed method

Co-authored-by: Jakub Trllo <43494761+iLLiCiTiT@users.noreply.github.com>

kalisp committed:
Reorganized flow

Co-authored-by: Jakub Trllo <43494761+iLLiCiTiT@users.noreply.github.com>

kalisp committed:
Used renamed method

iLLiCiTiT committed:
Merge branch 'develop' into 1488-yn-0117-usd-texture-publish-to-different-drive

iLLiCiTiT committed:
Merge pull request #1506 from BigRoy/1488-yn-0117-usd-texture-publish-to-different-drive

Use correct publish template in `get_instance_expected_output_path`

kalisp committed:
Merge branch 'develop' into enhancement/YN-0130_copy_folder_thumbnail

iLLiCiTiT committed:
Merge branch 'develop' into enhancement/lock-lower-pyobjc-core

iLLiCiTiT committed:
Merge pull request #1509 from ynput/enhancement/lock-lower-pyobjc-core

Chore: Lock pyobjc-core to 11.1

iLLiCiTiT committed:
Merge branch 'develop' into bugfix/support-older-ayon-launcher

iLLiCiTiT committed:
Merge pull request #1511 from ynput/bugfix/support-older-ayon-launcher

Chore: Support older AYON launcher

iLLiCiTiT committed:
Merge branch 'develop' into enhancement/YN-0130_copy_folder_thumbnail

iLLiCiTiT committed:
Merge pull request #1510 from ynput/enhancement/YN-0130_copy_folder_thumbnail

Library: Implemented copy of source folder thumbnail

ynbot committed:
[Automated] Add generated package files from main

ynbot committed:
[Automated] Update version in package.py for develop

github-actions[bot] committed:
chore(): update bug report / version

rdelillo committed:
Fix legacy OTIO clips detection on range remap.

rdelillo committed:
Add comment.

jakubjezek001 committed:
Merge branch 'develop' into bugfix/YN-0080-frame-range-is-showing-wrong-in-publisher

jakubjezek001 committed:
Merge pull request #1498 from ynput/bugfix/YN-0080-frame-range-is-showing-wrong-in-publisher

Corrects file sequence frame offset

FuzzkingCool committed:
Add double-click functionality to open thumbnails in LoaderWindow

FuzzkingCool committed:
Enhancement: Add video playback support in ThumbnailPainterWidget and LoaderWindow

- Implemented video representation path retrieval in LoaderWindow.
- Enhanced ThumbnailPainterWidget to detect video files and manage playback.
- Added functionality to extract the first frame from videos for thumbnail previews.
- Updated mouse event handling to toggle video playback on single click.

MustafaJafar committed:
Merge branch 'develop' into loader_video_representation_thumbnail_playback

MustafaJafar commented:
> This sounds neat! Any chance you can attach some previews of what it looks like in action?


https://github.com/user-attachments/assets/902b55a8-07cf-4dbb-9afa-0d4a29b8a127

---

There are few things to consider, 
1. Empty thumbnails follow the type of the last working thumbnail. and I believe they all should default to look like an empty image not empty video.
2. The video thumbnail doesn't work in place inside the loader.

FuzzkingCool commented:
Thanks @MustafaJafar for the video and feedback. These are definitely important issues to fix before this can get merged. Odd that the single-click play isn't working but I will try and recreate and get back to you. Cheers!

FuzzkingCool committed:
Implement auto-selection of the latest product item and improve thumbnail handling in LoaderWindow

- Added a method to select the latest published item in the ProductsWidget.
- Enabled automatic selection of the first item when no versions or folders are selected in LoaderWindow.
- Improved thumbnail management by clearing thumbnails when no entities are selected or when clearing current thumbnails.

FuzzkingCool committed:
Merge branch 'loader_video_representation_thumbnail_playback' of https://github.com/FuzzkingCool/ayon-core into loader_thumbnail_open_or_play_video

iLLiCiTiT commented on client/ayon_core/tools/loader/ui/window.py:
```diff
@@ -660,3 +764,26 @@ def _on_folders_refresh(self):
+            and not self._selected_folder_ids
+        ):
+            self._auto_select_first_item = False
+            self._products_widget.select_latest_item()
```
So if I select different version and then hit refresh, it will also change what I had selected in browser?

iLLiCiTiT commented on client/ayon_core/tools/loader/ui/window.py:
```diff
@@ -660,3 +764,26 @@ def _on_folders_refresh(self):
+            self._auto_select_first_item = False
+            self._products_widget.select_latest_item()
+
+    def _on_thumbnail_double_clicked(self, thumbnail_path):
```
Please move to controller.

iLLiCiTiT commented on client/ayon_core/tools/utils/thumbnail_paint_widget.py:
```diff
@@ -33,19 +47,136 @@ def __init__(self, parent):
+                self._on_media_status_changed
+            )
+        except (AttributeError, TypeError):
+            # Qt5 fallback
```
What exactly is different? And why it crashes? The only difference I see is `QMediaPlayer` initialization (one line) but whole logic is duplicated. If it's because `VideoSurface` should be used then do


```python
if hasattr(QtMultimedia.QMediaPlayer, "VideoSurface")"
    self._media_player = QtMultimedia.QMediaPlayer(
        self, QtMultimedia.QMediaPlayer.VideoSurface
    )
else:
    self._media_player = QtMultimedia.QMediaPlayer(self)

iLLiCiTiT commented on client/ayon_core/tools/utils/thumbnail_paint_widget.py:
```diff
@@ -33,19 +47,136 @@ def __init__(self, parent):
+        if not filepath:
+            return False
+        basename = os.path.basename(filepath).lower()
+        # Check for h264_* naming convention
```
Why?

iLLiCiTiT commented on client/ayon_core/tools/utils/thumbnail_paint_widget.py:
```diff
@@ -33,19 +47,136 @@ def __init__(self, parent):
+        if "h264_" in basename or "h264" in basename:
+            return True
+        # Check for common video extensions
+        video_exts = (".mp4", ".mov", ".avi", ".mkv", ".webm", ".m4v")
```
Should be constant and can be set.
```suggestion
        video_exts = {".mp4", ".mov", ".avi", ".mkv", ".webm", ".m4v"}
```
